cmake/KokkosKernels_Version_Info.hpp

```diff
@@ -0,0 +1,34 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOSKERNELS_GIT_VERSION_INFO_H
+#define KOKKOSKERNELS_GIT_VERSION_INFO_H
+
+#include <string>
+
+namespace KokkosKernels {
+namespace Impl {
+
+extern std::string GIT_BRANCH;
```

Comment:
Why not constexpr variables?

Reply:
Possible if we switch to header only version.

Reply:
Switched to `constexpr std::string_view`.
`constexpr std::string` requires C++20.

Reply:
Yes string_view is appropriate here, thanks @cz4rs